internal/sveltos/profile.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sveltos
+
+import (
+	"context"
+	"fmt"
+
+	hmc "github.com/Mirantis/hmc/api/v1alpha1"
+	"github.com/go-logr/logr"
+	sveltosv1beta1 "github.com/projectsveltos/addon-controller/api/v1beta1"
+	libsveltosv1beta1 "github.com/projectsveltos/libsveltos/api/v1beta1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+)
+
+type ReconcileProfileOpts struct {
+	OwnerReference *metav1.OwnerReference
+	HelmChartOpts  []HelmChartOpts
+}
+
+type HelmChartOpts struct {
+	RepositoryURL         string
+	RepositoryName        string
+	ChartName             string
+	ChartVersion          string
+	ReleaseName           string
+	ReleaseNamespace      string
+	Values                *apiextensionsv1.JSON
+	PlainHTTP             bool
+	InsecureSkipTLSVerify bool
+}
+
+// ReconcileProfile reconciles a Sveltos Profile object.
+func ReconcileProfile(ctx context.Context,
+	cl client.Client,
+	l logr.Logger,
+	namespace string,
+	name string,
+	matchLabels map[string]string,
+	opts ReconcileProfileOpts,
+) (*sveltosv1beta1.Profile, controllerutil.OperationResult, error) {
```

Comment:
It looks like the `OperationResult` is unused and not needed here as a result now